.changeset/thin-crabs-care.md

```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': minor
+---
+
+Added 'ref' support for `Button`, `UnstyledButton`, `Page.secondaryActions`, `ActionList.items`.
```

Comment:
Hi @knames !
I just tested your changes on the Product Detail page for the Duplicate button and they work great! 👍 

But I couldn't implement ref for grouped actions, getting a type error: `Object literal may only specify known properties, and 'ref' does not exist in type 'MenuGroupDescriptor'`
Can we add 'ref' to type 'MenuGroupDescriptor'`?  I created draft PR explaining the test case [here](https://github.com/Shopify/web/pull/87281/files#r1145441168)

Reply:
@knames Just tested your updates. After I provided an` id `as well as a `ref ` on the action which is within grouped action - it no longer closes "More action", but the focus does not return to its activator. Maybe I implemented it in the wrong way?
My commit: https://github.com/Shopify/web/pull/87281/commits/48258a6d75b517b8c267de4561aecc09aa6d242f 
My Spin: https://review-ks-button-ref.oksana-azarova.us.spin.dev/
Video how it works: 

https://user-images.githubusercontent.com/104942025/227372589-1836a4f4-591f-4422-aecb-a3e35c649192.mov

CC: @tatianau

Reply:
@knames  I also tried setting it up on my spin instance with the Discounts `Export` button in the `DiscountList`.

The component already has a `ref` created and passed to the `ExportModal` as an `activator`, the only thing I added is passing the `ref` and `id` to `secondaryActions` and [publishing your polaris changes to web with yalc](https://web.docs.shopify.io/docs/guides/working-with-polaris). 
- [Spin](https://shop1.shopify.modal-focus-polaris.tatiana-uspenskaia.us.spin.dev/admin/discounts)
- [Consumer changes in web](https://github.com/Shopify/web/commit/d5005b4cb6c4239d89e505b64a9347c1c865d307)

The focus works on the standalone `Export` button but it doesn't work when collapsed as it just closes the popover altogether 🤔 

https://user-images.githubusercontent.com/15054990/227380979-ca153633-a92a-4bec-87c3-6b6229eb68c4.mp4

Reply:
Hey @oksanashopify! You'll need to install the snapshot release for this PR's changes on your Shopify/web spinstance so you can test: 

```bash
yarn add @shopify/polaris@0.0.0-snapshot-release-20230323222520
```

Reply:
Just looked at your PR and [this line](https://github.com/Shopify/web/pull/87281/files#r1146974853) is all that needs to change 😁

Reply:
Thank you @chloerice, unfortunately that [did not resolve the issue](https://github.com/Shopify/web/pull/87281#discussion_r1147692639).